FEXHeaderUtils/FEXHeaderUtils/RingBuffer.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+#include <bit>
+#include <cstddef>
+
+namespace FHU {
+// This is a fast thread-local non-blocking ring-buffer.
+// Very useful for debugging purposes to see the history of something.
+template<typename T, size_t Elements>
+requires (std::popcount(Elements) == 1 && std::is_trivially_copyable_v<T>)
+class NonBlockRingBuffer final {
```

Comment:
I'd have reservations against this in production code. Since you're adding it for the purposes of debugging specifically, could you make this `[[deprecated("Not for production use")]]` to emphasize the intended use case?

Reply:
Sure, added.